src/Symfony/Component/Form/Extension/Validator/Constraints/FormValidator.php

```diff
@@ -142,6 +145,54 @@ public function validate($form, Constraint $formConstraint)
         }
     }
 
+    /**
+     * Filter out form filed violations to meet the requirements of the sequence of groups.
+     *
+     * This is necessary because each form field is validated independently.
+     */
+    private function filterFormFieldsGroupSequenceViolations(GroupSequence $groupSequence)
+    {
+        if (\count($violations = $this->context->getViolations()) < 2) {
+            return;
+        }
+
+        // collect violation offsets with an associated group
+        $violationGroups = [];
+
+        foreach ($violations as $offset => $violation) {
+            $violationGroups[$offset] = array_map(static function ($group) {
+                return $group;
+            }, $violation->getConstraint()->groups);
+        }
+
+        // collect groups that meet the requirements
+        $groupsToKeep = [];
+
+        foreach ($groupSequence->groups as $seqGroups) {
+            $seqGroups = !\is_array($seqGroups) ? [$seqGroups] : $seqGroups;
+
+            // if violations have groups in the current groups of the sequence,
+            // keep this current groups to remove all other violations that don't belong this groups later
```

Comment:
belong to this

Reply:
rewrite master comment, remove all others